### PR TITLE
Monthly Fee for Recurring Session

### LINKED ISSUE #121: Calculate Monthly Fees for 1 student for recurring session

### PR DESCRIPTION
**Pull request by enhao25**
Resolves #121.

Completed

### CODE REVIEW

**Review by nowknowing — APPROVED** [review]
LGTM!

**Review by nowknowing — APPROVED** [review]
LGTM!
utterly minute also debatable so ignore if dont think so, but getFee might benefit from being inclusive on both inputs. looks like its change < to <=. oh but also the caller

**Comment by enhao25** [comment]
> LGTM!
> utterly minute also debatable so ignore if dont think so, but getFee might benefit from being inclusive on both inputs. looks like its change < to <=. oh but also the caller 

Ok ah. The reason why I didn't make it inclusive end is because it is easier to get the months fee. For example to get Mar fee, I can put 1 Mar start, and then plus 1 month to 1 Mar to get 1 Apr as the end. 

If it is inclusive, I will have to do it such that the end period is the end of the month, but then some ends at 28, 29, 30 or 31. So its much harder to do that. So for the purpose of monthly fee, I would tink this way is better.

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -201,13 +203,38 @@ public double getFee(LocalDateTime startPeriod, LocalDateTime endPeriod) {
...
+        // Minus one because the end date for numOfSessionBetween is inclusive
+        SessionDate endDate = new SessionDate(endPeriod.minusDays(1));
+        int numOfSession = recurringSession.numOfSessionBetween(startDate, endDate);
+
+        // Ensures that there is more than 0 session, so we can get the fees accordingly
+        if (numOfSession > 0) {
+            System.out.println("Fee: " + numOfSession);
+            return recurringSession.getFee().getFee() * numOfSession;
+        }
+        return 0;
```
Comment (chwoozy) [inline src/main/java/seedu/address/model/ModelManager.java]: Do you want to throw an error if `numOfSession` is less than 1 instead, easier for debugging in the future.

**Review by chwoozy — APPROVED** [review]
Added some comments, otherwise LGTM! 😄

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (enhao25) [reply src/main/java/seedu/address/model/ModelManager.java]: Haha ok! This code is moved to feeUtil, but then hearing ur advice, I have put an assertion that numOfSession >= 0 (Sometimes can be 0 depending on the period given). In this way, if we want to debug, we can just activate assertion, but then it wouldn't return an exception during normal activity.

**Pull request merged**